geom.go

```diff
@@ -43,6 +44,7 @@ func (p Point) Length() float64 {
 	return math.Sqrt(p.X*p.X + p.Y*p.Y)
 }
 
+// A Rectangle contains the points with Min.X <= X <= Max.X, Min.Y <= Y <= Max.Y.
```

Comment:
I'm not 100% sure if it's `<= Max.X` and `<= Max.Y`. Please comment here with how you verified that. Sorry I can't do this myself now, will try when I find time if you don't.